icu4c/source/data/brkitr/ja.txt

```diff
@@ -238,4 +238,782 @@ ja{
+        "BW1:たい:360",
+        "UW4:使:217",
+        "UW2:き:130",
+        "TW4:かなり:473",
+        "UB6:109:-41",
+        "BB1:108108:-23",
+        "UW4:込:-340",
+        "TW3:と言っ:-530",
+        "UW6:だ:-69",
+        "UW5:り:-71",
+        "UW5:よ:-115",
+        "BW3:どう:297",
+        "UW4:…:-240",
+        "UW3:や:229",
+        "BW1:かし:507",
+        "BW3:かっ:-348",
+        "UW4:今:171",
+        "UW3:『:-320",
+        "UW4:思:239",
+        "UB2:109:16",
+        "UW4:く:-195",
+        "UW3:京:-277",
+        "UW6:ー:-41",
+        "UW1:ん:69",
+        "BW1:うな:280",
+        "TB2:108107107:-264",
+        "UW1:と:30",
+        "TB4:108109108:249",
+        "TB2:107108108:-97",
+        "BW1:この:-163",
+        "BW2:ので:-221",
+        "UW4:み:96",
+        "UW5:わ:83",
+        "UW6:や:82",
+        "BW1:れて:-218",
+        "UW2:や:-93",
+        "UW6:こ:-53",
+        "UW4:な:40",
+        "UW5:め:28",
+        "BW1:もう:285",
+        "TB4:120108120:27",
+        "BW1:より:283",
+        "UW4:合:-211",
+        "UW6:け:-92",
+        "BW1:少し:214",
+        "BW2:でし:-225",
+        "UW4:と:-54",
+        "TB1:109109109:53",
+        "UW3:ー:105",
+        "BW2:くな:-198",
+        "UW2:く:-53",
+        "UW2:我:-277",
+        "BW2:いも:198",
+        "BW3:わか:184",
+        "TB2:120109120:-264",
+        "UW4:も:-106",
+        "UW1:あ:14",
+        "UW4:最:185",
+        "BW1:るの:-155",
+        "UW2:全:185",
+        "UW6:０:106",
+        "UW4:放:-119",
+        "UW4:京:53",
+        "BW3:かけ:208",
+        "UW2:少:92",
+        "BW3:もう:262",
+        "UW2:多:106",
+        "UW2:う:-52",
+        "TB1:108108072:105",
+        "UW1:を:-25",
+        "UW3:光:-79",
+        "BW1:！！:104",
+        "UW2:ャ:141",
+        "BW3:すぐ:129",
+        "UW4:帯:-114",
+        "UW6:し:26",
+        "BW3:でも:64",
+        "BW2:、そ:-113",
+        "TB3:120162162:26",
+        "TB2:109108120:77",
+        "UW3:わ:-64",
+        "UB4:109:13",
+        "TB4:120120120:13",
+        "UW5:都:26",
+        "UW5:ず:89",
+        "UW2:バ:115",
+        "UW2:京:-49",
+        "UW3:ゃ:89",
+        "BW1:い、:-114",
+        "BW3:よく:51",
+        "BW1:たら:64",
+        "BW2:のよ:-64",
+        "UW2:思:-51",
+        "BW1:うに:-38",
+        "BW1:の間:89",
+        "UW6:ん:13",
+        "UW6:ず:-64",
+        "BW1:った:13",
+        "TW3:ること:-48",
+        "BW3:とて:76",
+        "TW1:ような:63",
+        "UW6:ぱ:62",
+        "TB3:109120108:13",
+        "TW4:って、:112",
+        "TW4:なんて:-76",
+        "TW2:その後:-50",
+        "UW6:ら:-13",
+        "TW4:ことに:-49",
+        "UW3:＞:63",
+        "TW3:てしま:-50",
+        "UW3:い:13",
+        "TB4:120108107:13",
+        "UW2:ひ:-50",
+        "UW6:め:24",
+        "UW6:で:-12",
+        "BW3:なる:24",
+        "UW5:ご:12",
+        "BW2:りし:24",
+        "UW6:電:12",
+        "UW1:は:-12",
+        "BW1:いも:-24",
+        "BW3:すご:12",
+        "UW4:通:-12",
+        "BW3:おり:-12",
+        "BW3:かか:12",
+        "BW1:思い:-12",
+    }
+    unicode_blocks {
+        0,
+        128,
+        256,
+        384,
```

Comment:
What are these? Are they the start code points of the Unicode Blocks?

Why are you using Unicode Blocks? They are an artifact of how the organization manages its code space, but they are not generally useful for processing. (Some people think they want to use the Block property but they are better served by the Script or Script_Extensions property.)

ICU has data and API for which code point is in which Unicode Block.

If you really wanted and needed to put this kind of data into a resource bundle, you should store it as integers, not as strings. Resource bundles have an intvector value type. (But as I said above, if these are Unicode Blocks, you need not add storage for them at all.)

Reply:
Yes, they're the start code points of the Unicode blocks. Unicode block is used by the machine learning model as a clue to decide where to segment the sentence.
If the data is already available as an API, it should be fine to use that.

Reply:
> Yes, they're the start code points of the Unicode blocks. Unicode block is used by the machine learning model as a clue to decide where to segment the sentence.

Ok. FYI, usually the script property is more useful. For example, several scripts have characters in multiple blocks.

> If the data is already available as an API, it should be fine to use that.

`UBlockCode ublock_getCode(UChar32 c)`
https://unicode-org.github.io/icu-docs/apidoc/released/icu4c/uchar_8h.html#a2e70316e53f6fa0bdc7fb29d70671076

`UScriptCode uscript_getScript(UChar32 codepoint, UErrorCode *err)`
https://unicode-org.github.io/icu-docs/apidoc/released/icu4c/uscript_8h.html#afb6f615133cb5725fcebae41b294afe0

Reply:
The design is to get UnicodeBlock index.
Shuhei's index follows this https://www.unicode.org/Public/14.0.0/ucd/Blocks.txt
ICU's index is defined in https://github.com/unicode-org/icu/blob/main/icu4c/source/common/unicode/uchar.h
Since they don't match, I still use the original design.
For example, Hiragana's index is 108 in Blocks.txt but 62 in uchar.h

Reply:
Well, the blocks are the same, but Unicode doesn't prescribe what "index" number each block should have. In other words, this is just a permutation -- the same data with a different numbering. We should either look at which blocks the BudouX code looks for, and use the corresponding ICU constants, or, if the block numbers are baked into the model, then probably add a little permutation table. The latter is probably too much work for now, so we need to remember a TODO for it.

Reply:
I replied in another thread, but BudouX looks for the Unicode blocks below.

- General Punctuation
- Miscellaneous Symbols
- CJK Symbols and Punctuation
- Hiragana
- Katakana
- CJK Unified Ideographs
- Halfwidth and Fullwidth Forms